Core/include/Acts/TrackFinding/CombinatorialKalmanFilter.hpp

```diff
@@ -1427,19 +1440,21 @@ class CombinatorialKalmanFilter {
                  << combKalmanResult.result.error().message()
                  << " with the initial parameters: \n"
                  << initialParameters.parameters());
-      return combKalmanResult.result.error();
     }
 
     std::vector<typename TrackContainer::TrackProxy> tracks;
 
     for (auto tip : combKalmanResult.lastMeasurementIndices) {
       auto track = trackContainer.getTrack(trackContainer.addTrack());
       track.tipIndex() = tip;
-      const BoundTrackParameters& parameters =
-          combKalmanResult.fittedParameters.find(tip)->second;
-      track.parameters() = parameters.parameters();
-      track.covariance() = *parameters.covariance();
-      track.setReferenceSurface(parameters.referenceSurface().getSharedPtr());
+
+      if (auto it = combKalmanResult.fittedParameters.find(tip);
+          it != combKalmanResult.fittedParameters.end()) {
```

Comment:
So here we'll get tracks out that have random parameters and covariances, and no reference surface. Should we even create a track for these cases?

Reply:
true would be best to not create these tracks at all